src/main/java/com/puppycrawl/tools/checkstyle/DetailASTImpl.java

```diff
@@ -0,0 +1,406 @@
+        while (ast != null && ast.getNextSibling() != null) {
+            ast = ast.getNextSibling();
+        }
+        return ast;
+    }
+
+    /**
+     * Finds column number in the first non-comment node.
+     *
+     * @param ast DetailAST node.
+     * @return Column number if non-comment node exists, -1 otherwise.
+     */
+    private static int findColumnNo(DetailAST ast) {
+        int resultNo = -1;
+        DetailAST node = ast;
+        while (node != null) {
+            // comment node can't be start of any java statement/definition
+            if (TokenUtil.isCommentType(node.getType())) {
+                node = node.getNextSibling();
+            }
+            else {
+                resultNo = node.getColumnNo();
+                break;
+            }
+        }
+        return resultNo;
+    }
+
+    /**
+     * Finds line number in the first non-comment node.
+     *
+     * @param ast DetailAST node.
+     * @return Line number if non-comment node exists, -1 otherwise.
+     */
+    private static int findLineNo(DetailAST ast) {
+        int resultNo = -1;
+        DetailAST node = ast;
+        while (node != null) {
+            // comment node can't be start of any java statement/definition
+            if (TokenUtil.isCommentType(node.getType())) {
+                node = node.getNextSibling();
+            }
+            else {
+                resultNo = node.getLineNo();
+                break;
+            }
+        }
+        return resultNo;
+    }
+
+    /**
+     * Returns token type with branch.
+     * @return the token types that occur in the branch as a sorted set.
+     */
+    private BitSet getBranchTokenTypes() {
+        // lazy init
+        if (branchTokenTypes == null) {
+            branchTokenTypes = new BitSet();
+            branchTokenTypes.set(getType());
+
+            // add union of all children
+            DetailASTImpl child = getFirstChild();
+            while (child != null) {
+                final BitSet childTypes = child.getBranchTokenTypes();
+                branchTokenTypes.or(childTypes);
+
+                child = child.getNextSibling();
+            }
+        }
+        return branchTokenTypes;
+    }
+
+    @Override
+    public boolean branchContains(int type) {
+        return getBranchTokenTypes().get(type);
+    }
+
+    @Override
+    public DetailAST getPreviousSibling() {
+        return previousSibling;
+    }
+
+    @Override
+    public DetailAST findFirstToken(int type) {
+        DetailAST returnValue = null;
+        for (DetailAST ast = getFirstChild(); ast != null; ast = ast.getNextSibling()) {
+            if (ast.getType() == type) {
+                returnValue = ast;
+                break;
+            }
+        }
+        return returnValue;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getText(), getType(), getLineNo(), getColumnNo());
+    }
+
+    @Override
+    public boolean equals(Object object) {
```

Comment:
This is a new method because we are required to give access this for the 2 checks I recently changed to remove antlr support.

Reply:
This was removed.